orsopy/fileio/base.py

```diff
@@ -0,0 +1,174 @@
+"""
+Implementation of the base classes for the ORSO header.
+"""
+
+# author: Andrew R. McCluskey (arm61)
+
+import yaml
+
+
+def noop(self, *args, **kw):
+    pass
+
+
+yaml.emitter.Emitter.process_tag = noop
+
+
+def _repr(class_to_represent):
+    """
+    The representation object for all the Header sub-classes. This returns a
+    string in a yaml format which will be ORSO compatible.
+
+    Args:
+        class_to_represent (:py:class:`object`): The class to be represented.
+
+    Returns:
+        (:py:attr:`str`): A string representation.
+    """
+    cleaned_data = dict(
+        (k, v) for (k, v) in class_to_represent.__dict__.items(
+        ) if v is not None
+    )
+    return yaml.dump(cleaned_data, sort_keys=False)
```

Comment:
Shouldn't the representation be Pyhon-like. So something like "Class(attr1=value, attr2=value2)".

For the dump to yaml I would use a "dump" method (with possible keyword parameters).